AdyenUIHostUITests/Components/Affirm/AffirmComponentUITests.swift

```diff
@@ -0,0 +1,266 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import XCTest
+@testable import Adyen
+@testable import AdyenComponents
+
+class AffirmComponentUITests: XCTestCase {
+
+    private var paymentMethod: PaymentMethod!
+    private var apiContext: APIContext!
+    private var style: FormComponentStyle!
+    private var sut: AffirmComponent!
+    private let app = XCUIApplication()
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        print(app.debugDescription)
+        paymentMethod = AtomePaymentMethod(type: .atome, name: "Affirm")
+        apiContext = Dummy.context
+        style = FormComponentStyle()
+        sut = AffirmComponent(paymentMethod: paymentMethod,
+                              apiContext: apiContext,
+                              configuration: AffirmComponent.Configuration(style: style))
+    }
+
+    override func tearDownWithError() throws {
+        paymentMethod = nil
+        apiContext = nil
+        style = nil
+        sut = nil
+        try super.tearDownWithError()
+    }
+
+    func testSubmitForm_shouldCallDelegateWithProperParameters() throws {
+        // Given
+        let sut = AffirmComponent(paymentMethod: paymentMethod,
+                                  apiContext: apiContext,
+                                  configuration: AffirmComponent.Configuration(style: style))
+        let delegate = PaymentComponentDelegateMock()
+        sut.delegate = delegate
+        let expectedBillingAddress = PostalAddressMocks.newYorkPostalAddress
+        let expectedDeliveryAddress = PostalAddressMocks.losAngelesPostalAddress
+        UIApplication.shared.mainKeyWindow?.rootViewController = sut.viewController
+
+        // Then
+        let didSubmitExpectation = expectation(description: "PaymentComponentDelegate must be called when submit button is clicked.")
+        delegate.onDidSubmit = { data, component in
+            XCTAssertTrue(component === sut)
+            let details = data.paymentMethod as! AffirmDetails
+            XCTAssertEqual(details.shopperName?.firstName, "Katrina")
+            XCTAssertEqual(details.shopperName?.lastName, "Del Mar")
+            XCTAssertEqual(details.telephoneNumber, "2025550146")
+            XCTAssertEqual(details.emailAddress, "katrina@mail.com")
+            XCTAssertEqual(details.billingAddress, expectedBillingAddress)
+            XCTAssertEqual(details.deliveryAddress, expectedDeliveryAddress)
+            
+            sut.stopLoadingIfNeeded()
+            didSubmitExpectation.fulfill()
+        }
+
+        wait(for: .seconds(1))
```

Comment:
We recently reduced this number to 300 miliseconds on other tests. If 1 sec isn't necessary here, let's align with 300 ms.

Reply:
This I have modified.